### ISSUE TITLE
[Advice] Determine if proper DLL loaded into process. 

### ISSUE THREAD

**Issue by r3538987** [kcat/openal-soft#1055]
As audio in general more subjective, and varies per person, detecting if your configured setup loaded correctly may require additional checks. 

**What would be your way to visually see that your copied files into folder got actually loaded in.**

Setup:
Windows 10 22H2, with OpenAL installed. 

Here is sample, which started my doubts (via SystemInformer) :
Splinter Cell 1
![image](https://github.com/user-attachments/assets/d2c5881b-3bcf-4c23-9dff-5c8f672e4476)
Splinter Cell 3 Chaos Theory
![image](https://github.com/user-attachments/assets/45d41278-8b0c-473c-a3e2-9d46ad2356ed)

At some point I did applied EAX_Restore registry fix, which added dsound.dll keys for few paths. https://i.imgur.com/78HrPFP.png
After tinkering with Creative ALchemy, seems it added few entries (ones from %SystemRoot%). 

Now I'm looking into first screenshot, and there is two dsound.dll files mentioned, and one is from system32. This makes me think, that maybe whole EAX restoring procedure doesn't work on my end, because wrong file is loaded in.

**Comment by ThreeDeeJay** [comment]
IDrinkLava didn't mention issues loading DSOAL's dsound.dll from the game folder in [his video](https://youtu.be/5OHXsb0ZsC0).
So maybe you missed some registry keys from here: https://www.indirectsound.com/registryIssues.html
I made a script to automate the process, which includes even more keys here: https://github.com/kcat/dsoal/issues/34#issuecomment-1139239755

[There are rare cases of games that seem to hardcode the DLL path so they might load the system one](https://github.com/multitheftauto/mtasa-blue/issues/2687), but the tweak above should work with most.

As for checking which DLL are being loaded by the game, I use [API Monitor](http://www.rohitab.com/apimonitor), which is also useful to see if/which API calls are being made. 
![image-1](https://github.com/user-attachments/assets/08e071a5-56df-4b82-a575-d484fd6fd6a5)

**Comment by mirh** [comment]
You are probably getting different results because you are running one of the two games as an administrator.
But isn't the log file just enough to know whether it worked or not?

Sooner or later it would be cool if somebody could make a 2.0 registry fix with the auto-elevating batch magic.

**Comment by ThreeDeeJay** [comment]
> EDIT: actually it may even be that one game uses eax unified and the other doesn't

@mirh Apparently the both do
https://steamdb.info/depot/13561/
![image](https://github.com/user-attachments/assets/42f66e4b-38e4-47b2-8733-e1f2444fe20d)
https://steamdb.info/depot/13571/
![image](https://github.com/user-attachments/assets/aad8b1a1-d9f0-4030-aad2-3b21a458a2c8)

> p.s. sooner or later it would be cool if somebody could make a 2.0 registry fix with the auto-elevating batch magic.

Wait, did someone find which registry keys to patch to fix EAX/DS3D wrapper loading in those few pesky games that still loads the system dsound.dll from the system even after using [my script](https://github.com/kcat/dsoal/issues/34#issuecomment-1139239755)?

**Comment by mirh** [comment]
No, because if a game explicitly asks for a specific absolute path that's it (minus perhaps the [dllinject](https://github.com/kcat/dsoal/issues/51) hypothesis that still wasn't tested.. and AppInit_DLLs maybe, but I don't even want to think to it). Even though I suppose then it should be fairly easy to hex patch them.

I just meant you adding the [reg_own](https://github.com/kcat/dsoal/issues/34#issuecomment-1139602846) magic to the .bat, so that it's pretty much just a double click to run.
